ipn/ipnlocal/local.go

```diff
@@ -800,7 +803,44 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 		return
 	}
 
+	// Handle node expiry in the netmap
+	var nextExpiry time.Time
+	if st.NetMap != nil {
+		b.em.flagExpiredPeers(st.NetMap)
+
+		// Figure out when the next node in the netmap is expiring so we can
+		// start a timer to reconfigure at that point.
+		for _, peer := range st.NetMap.Peers {
+			if peer.KeyExpiry.IsZero() {
+				continue // tagged node
+			}
+			if peer.KeyExpiry.After(nextExpiry) {
+				nextExpiry = peer.KeyExpiry
```

Comment:
I don't follow this bit. Don't you want nextExpiry to be the soonest expiring one, not the latest?

Reply:
Gooooooood catch, yes; I got this backwards 🤦